### ISSUE TITLE
Empty :key + v-for breaks eslint vue

### ISSUE THREAD

**Issue by HerringtonDarkholme** [vuejs/eslint-plugin-vue#37]
Version:
eslint-plugin-vue 3.1.1
eslint: 4.0.0

Minimal template:

```vue
<template>
    <div v-for="(item, index) in suggestions" :key></div>
</template>
```

Expect:
report `v-bind` directive needs an attribute value.

Actual:
crash.

```
Cannot read property 'references' of null
TypeError: Cannot read property 'references' of null
    at isUsingIterationVar (node_modules/eslint-plugin-vue/lib/rules/no-invalid-v-for.js:24:32)
    at EventEmitter.VAttribute[directive=true][key.name='for'] (node_modules/eslint-plugin-vue/lib/rules/no-invalid-v-for.js:54:32)
    at emitOne (events.js:120:20)
    at EventEmitter.emit (events.js:210:7)
    at NodeEventGenerator.applySelector (node_modules/eslint/lib/util/node-event-generator.js:265:26)
    at NodeEventGenerator.applySelectors (node_modules/eslint/lib/util/node-event-generator.js:294:22)
    at NodeEventGenerator.enterNode (node_modules/eslint/lib/util/node-event-generator.js:308:14)
    at traverse (node_modules/eslint-plugin-vue/node_modules/vue-eslint-parser/lib/traverse-nodes.js:137:15)
    at traverse (node_modules/eslint-plugin-vue/node_modules/vue-eslint-parser/lib/traverse-nodes.js:146:21)
    at traverse (node_modules/eslint-plugin-vue/node_modules/vue-eslint-parser/lib/traverse-nodes.js:151:13)
```

**Comment by mysticatea** [comment]
Thank you for the report.

I confirmed it.

**Comment by michalsnik** [comment]
Released as `v3.1.2` 🚀
